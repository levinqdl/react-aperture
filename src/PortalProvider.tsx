import React from "react"
import SlotContext, { ISlotContext } from "./SlotContext"

const { Provider } = SlotContext

class PortalProvider extends React.Component<{}, ISlotContext> {
  counter: number = 0
  addSlot = (element: HTMLDivElement) => {
    const key = `s${this.counter++}`
    this.setState(({ slots }) => ({
      slots: {
        ...slots,
        [key]: element,
      },
    }))
    return key
  }
  state: ISlotContext = {
    slots: {},
    addSlot: this.addSlot,
  }
  render() {
    const { children } = this.props
    return <Provider value={this.state}>{children}</Provider>
  }
}
export default PortalProvider
