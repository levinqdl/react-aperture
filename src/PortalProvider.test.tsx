import React from "react"
import PortalProvider from "./PortalProvider"
import { render } from "react-testing-library"

describe("PortalProvider", () => {
  it("renders children", () => {
    const { container, getByText } = render(
      <PortalProvider>
        <div>Hello Portal Layout!</div>
      </PortalProvider>
    )
    getByText("Hello Portal Layout!")
    const layout = container.firstChild as HTMLElement
    expect(layout).toMatchSnapshot()
  })
})
