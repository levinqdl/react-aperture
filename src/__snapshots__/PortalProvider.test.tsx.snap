// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortalProvider renders children 1`] = `
<div>
  Hello Portal Layout!
</div>
`;
